common/src/main/scala/org/apache/celeborn/common/metrics/source/ThreadPoolSource.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.metrics.source
+
+import java.util.concurrent.{ExecutorService, ThreadPoolExecutor}
+
+import com.codahale.metrics.Gauge
+
+import org.apache.celeborn.common.CelebornConf
+
+class ThreadPoolSource(
+    conf: CelebornConf,
+    role: String)
+  extends AbstractSource(conf, role) {
+  override val sourceName = "ThreadPool"
+  def registerSource(
+      threadPoolName: String,
+      threadPoolExecutor: ThreadPoolExecutor): Unit = {
+    val label = Map("threadPool" -> threadPoolName)
+    addGauge(
+      "active_count",
+      label,
+      new Gauge[Long] {
+        override def getValue: Long = {
+          threadPoolExecutor.getActiveCount
+        }
+      })
+    addGauge(
+      "completed_task_count",
+      label,
+      new Gauge[Long] {
+        override def getValue: Long = {
+          threadPoolExecutor.getCompletedTaskCount
```

Comment:
Some of the invocations in this class, for example `getCompletedTaskCount`, will lock the threadpool, so we need be careful not to sample too frequently to avoid impact performance. I wonder if all these metrics are necessary, for example  `completed_task_count`, `remain_queue_capacity`.

Reply:
The frequency depending on user's poll interval setting .
completed_task_count and total_task_count  and remain_queue_capacity  is not necessary.

Reply:
PTAL again.